core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1189,12 +1189,23 @@ class KafkaApis(val requestChannel: RequestChannel,
                           properties: util.Properties = new util.Properties()): MetadataResponseTopic = {
     try {
       if (adminZkClient == null) {
-        throw new UnsupportedOperationException("Auto topic creation is not yet supported in KIP-500 mode")
+        if (replicationFactor < 1 || replicationFactor > Short.MaxValue) {
+          throw new IllegalArgumentException(s"Illegal replication factor: $replicationFactor")
+        }
+        val topics = new CreatableTopicCollection()
+        val configs = new CreateableTopicConfigCollection()
+        properties.forEach { (k, v) => configs.add(new CreateableTopicConfig().setName(k.toString).setValue(v.toString)) }
+        topics.add(new CreatableTopic().setName(topic).setNumPartitions(numPartitions).setReplicationFactor(replicationFactor.toShort).setConfigs(configs))
+        forwardingManager.channelManager.sendRequest(
+          new CreateTopicsRequest.Builder(new CreateTopicsRequestData().setTopics(topics)),
+          (_: ClientResponse) => {})
+        metadataResponseTopic(Errors.LEADER_NOT_AVAILABLE, topic, isInternal(topic), util.Collections.emptyList())
```

Comment:
shouldn't we wait until the topic creation succeeds, so that we match the earlier behavior?